docs/docs/code/python/README.md

```diff
@@ -0,0 +1,279 @@
+```
+
+Now this `pokemon` is accessible to downstream steps within `steps.code.pokemon`
+
+::: warning
+Not all data types can be stored in the `steps` data shared between workflow steps.  Only JSON serializable types can be passed, which includes:
+
+* lists 
+* dictionaries
+  :::
+
+## Using environment variables
+
+You can leverage any [environment variables defined in your Pipedream account](/environment-variables/#environment-variables) in a python step. This is useful for keeping your secrets out of code as well as keeping them flexible to swap API keys without having to update each step individually.
+
+To access them, use the `os` module.
+
+```python
+import os
+import requests
+
+token = os.environ['TWITTER_API_KEY']
+
+url = 'https://api.twitter.com/2/users/@pipedream/mentions'
+
+headers { 'Authorization': f"Bearer {token}"}
+r = requests.get(url, headers=headers)
+
+print(r.text)
+```
+
+:::tip
+There are 2 different ways of using the `os` module to access your environment variables.
+
+`os.environ['ENV_NAME_HERE']` will raise an error that stops your workflow if that key doesn't exist in your Pipedream account.
+
+Whereas `os.environ.get('ENV_NAME_HERE')` will _not_ throw an error and instead returns an empty string. 
+
+If your code relies on the presence of a environment variable, consider using `os.environ['ENV_NAME_HERE']` instead.
+:::
+
+## Handling errors
+
+You may find a need to exit a workflow early. In a Python step, just a `raise` an error to halt a step's execution.
+
+
+```python
+raise NameError('Something happened that should not. Exiting early.')
+```
+
+All exceptions from your Python code will appear in the logs area of the results.
+
+## File storage
+
+Not only can you run Python code in workflows, but you can also store files as well. This means you can upload or download photos, retrieve datasets or accept files from an HTTP request and much more.
+
+The `/tmp` directory is accessible from your workflow steps for saving and retrieving files.
+
+You have full access to read & write both files & directories in `/tmp`. 
+
+### Writing a file to /tmp
+
+```python
+import requests
+
+# download the Python logo
+r = requests.get('https://www.python.org/static/img/python-logo@2x.png')
+
+# create a new file python-logo.png in the /tmp/data directory
+with open('/tmp/python-logo.png', 'wb') as f:
+  # save the content of the HTTP response into the file
+  f.write(r.content)
+```
+
+Now `/tmp/python-logo.png` holds the official Python logo.
+
+### Reading a file from /tmp
+
+You can also open files you have previously stored in the `/tmp` directory. Let's open the `python-logo.png` we downloaded in the previous example.
+
+```python
+import os
+
+with open('/tmp/python-logo.png') as f:
+  # put the contents of the file into a variable
+  file_data = f.read()
+```
+
+### Listing files in /tmp
+
+If you need to check what files are currently in `/tmp` you can list them and print the results to the workflow console:
```

Comment:
"workflow console" -> "**Logs** section of **Results"** to keep the reference consistent.